libs/@blockprotocol/core/src/service-handler.ts

```diff
@@ -48,27 +63,56 @@ export abstract class ServiceHandler {
     serviceName,
     sourceType,
   }: {
-    element: HTMLElement;
+    element?: HTMLElement | null;
     serviceName: string;
     sourceType: "block" | "embedder";
   }) {
-    this.element = element ?? null;
     this.serviceName = serviceName;
     this.sourceType = sourceType;
 
+    if (element) {
+      this.registerService(element);
+    }
+  }
+
+  initialize(element: HTMLElement) {
```

Comment:
As this is a method that is exposed to end-users, I think we should document this – here's a starting point
```suggestion
  /**
   * You only need to use this if you are constructing a service directly.
   * You do not need to use it if you're using a React hook or block template.
   *
   * This initializes a service with the element it will listen for messages on,
   * and must be called if the service was constructed without an element.
   */
  initialize(element: HTMLElement) {
```